tools/samtools_view.cwl

```diff
@@ -0,0 +1,89 @@
+  exclude_flags: { type: 'string?', inputBinding: { position: 2, prefix: "--exclude-flags" }, doc: "Do not output alignments with any bits set in FLAG present in the FLAG field." }
+  include_flags: { type: 'string?', inputBinding: { position: 2, prefix: "--include-flags" }, doc: "Only output alignments with any bit set in FLAG present in the FLAG field." }
+  exclude_all: { type: 'string?', inputBinding: { position: 2, prefix: "-G" }, doc: "Do not output alignments with all bits set in INT present in the FLAG field." }
+  subsample: { type: 'float?', inputBinding: { position: 2, prefix: "--subsample" }, doc: "Output only a proportion of the input alignments, as specified by 0.0 ≤ FLOAT ≤ 1.0, which gives the fraction of templates/pairs to be kept" }
+  subsample_seed: { type: 'int?', inputBinding: { position: 2, prefix: "--subsample-seed" }, doc: "Subsampling seed used to influence which subset of reads is kept." }
+  add_flags: { type: 'string?', inputBinding: { position: 2, prefix: "--add-flags" }, doc: "Adds flag(s) to read." }
+  remove_flags: { type: 'string?', inputBinding: { position: 2, prefix: "--remove-flags" }, doc: "Remove flag(s) from read." }
+  remove_tag: { type: 'string?', inputBinding: { position: 2, prefix: "--remove-tag" }, doc: "Comma-separated read tags to strip" }
+  keep_tag: { type: 'string?', inputBinding: { position: 2, prefix: "--keep-tag" }, doc: "Comma-separated read tags to preserve" }
+  remove_b: { type: 'boolean?', inputBinding: { position: 2, prefix: "--remove-B" }, doc: "Collapse the backward CIGAR operation" }
+  no_pg: { type: 'boolean?', inputBinding: { position: 2, prefix: "--no-PG" }, doc: "Do not add a @PG line to the header of the output file." }
+  reference_fasta: { type: 'File?', secondaryFiles: [{ pattern: '.fai', required: false}], inputBinding: { position: 2, prefix: "--reference" }, doc: "Reference sequence FASTA FILE" }
+  write_index: { type: 'boolean?', inputBinding: { position: 2, prefix: "--write-index" }, doc: "Automatically index the output files" }
+  input_format: { type: 'string?', inputBinding: { position: 2, prefix: "--input-fmt" }, doc: "Specify input format (SAM, BAM, CRAM) and any other options/values FORMAT[,OPT[=VAL]]..." }
+  output_format: { type: 'string?', inputBinding: { position: 2, prefix: "--output-fmt" }, doc: "Specify output format (SAM, BAM, CRAM) and any other options/values FORMAT[,OPT[=VAL]]..." }
+  cpu:
+    type: 'int?'
+    default: 1
+    doc: "Number of CPUs to allocate to this task."
+    inputBinding:
+      prefix: "--threads"
+      position: 2
+  ram:
+    type: 'int?'
+    default: 4
+    doc: "GB size of RAM to allocate to this task."
+
+outputs:
+  output:
+    type: File
+    secondaryFiles: [{pattern: '.bai', required: false}, {pattern: '^.bai', required: false}, {pattern: '.crai', required: false}, {pattern: '^.crai', required: false}]
+    outputBinding:
+      glob: '*.*am'
```

Comment:
is this for *.bam and/or *.cram? Would this work for both?

Reply:
Yes, this will grab SAMs, BAMs, or CRAMs